app/controllers/application_controller.rb

```diff
@@ -5,6 +5,142 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    if is_logged_in
+      redirect '/tweets'
+    else
+      erb :homepage
+    end
+  end
+
+  get '/signup' do
+    if is_logged_in
+      redirect '/tweets'
+    else
+      erb :"tweets/users/create_user"
+    end
+  end
+
+  post '/signup' do
+      if params[:username] != "" && params[:email] != "" && params[:password] != ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.